upup/pkg/fi/nodeup/nodetasks/archive.go

```diff
@@ -158,7 +160,12 @@ func (_ *Archive) RenderLocal(t *local.LocalTarget, a, e, changes *Archive) erro
 			return fmt.Errorf("error creating directories %q: %v", targetDir, err)
 		}
 
-		args := []string{"tar", "xf", localFile, "-C", targetDir}
+		var extractArgs = "xf"
+
+		if e.Gzip {
+			extractArgs = "xzf"
```

Comment:
This is fine.  Some alternatives you might want to consider: What we did in assetstore.go is we look at the extension, and try to guess whether we should expand it: `if strings.HasSuffix(file, ".tar.gz") || strings.HasSuffix(file, ".tgz")`.  I also think `tar` can usually figure it out, but we might not want to rely on that.  Also because we are in nodeup and so we have the file at this point, we could detect the file type (e.g. run `file`).

But this is good enough for now, unless you're incline to change it!